lib/api/CInferenceModelDefinition.cc

```diff
@@ -380,6 +392,48 @@ std::string CInferenceModelDefinition::jsonString() {
     return resultString;
 }
 
+std::stringstream CInferenceModelDefinition::jsonStringCompressedFormat() const {
+    std::stringstream compressedStream;
+    using TFilteredOutput = boost::iostreams::filtering_stream<boost::iostreams::output>;
+    std::string modelDefinitionStr{jsonString()};
```

Comment:
How hard would it be to change all the `addToDocument` methods that relate to the trained model to instead write JSON directly to a `ostream`?  So instead of building up RapidJSON objects in memory that then get serialised all in one go, the fields get written one at a time?

This wouldn't be thread-safe for the `ostream` that was being written to, but that would be OK because it would be a compression stream that was only being used by one thread.  This would avoid holding all the RapidJSON objects in memory until the complete state was built up.  Instead, the compressed string would gradually be built, and at any time uncompressed JSON would only exist in tiny pieces in the compression buffer.